src/feature.js

```diff
@@ -356,6 +360,67 @@ var feature = function (arg) {
 
   ////////////////////////////////////////////////////////////////////////////
   /**
+   * Set style(s) from array(s).  For each style, the array should have one
+   * value per data item.  The values are not converted or validated.  Color
+   * values should be objects with r, g, b values on a scale of [0, 1].  If
+   * invalidate values are given the behavior is undefined.
+   *   For features where this._subcomponentStyles is an object and a style
+   * name is a key in that object, and the first entry of the styleArray is
+   * itself an array, then each entry of the array is expected to be an array,
+   * and values are used from these subarrays.  This allows a style to apply,
+   * for instance, per vertex of a data item rather than per data item.
+   *
+   * @param {string|object} keyOrObject: either the name of a single style or
+   *    an object where the keys are the names of styles and the values are
+   *    each arrays.
+   * @param {array} styleArray: if keyOrObject is a string, an array of values
+   *    for the style.  If keyOrObject is an object, this parameter is ignored.
+   * @param {boolean} refresh: true to redraw the feature when it has been
+   *    updated.  If an object with styles is passed, the redraw is only done
+   *    once.
+   * @returns {object} the feature
+   */
+  ////////////////////////////////////////////////////////////////////////////
+  this.updateStyleFromArray = function (keyOrObject, styleArray, refresh) {
+    if (typeof keyOrObject !== 'string') {
+      $.each(keyOrObject, function (key, value) {
+        m_this.updateStyleFromArray(key, value);
+      });
+    } else {
+      var fallback;
+      if (keyOrObject.toLowerCase().match(/color$/)) {
+        fallback = {r: 0, g: 0, b: 0};
```

Comment:
Sure I am fine with being here for now and we can revisit if needed to be in the future.